### PR TITLE
feat: 💄 responsive header and togglable nav in mobile screens

### LINKED ISSUE #35: Make Homepage Responsive/Adaptive to mobile
**Todo**

- [x] Design for the mobile version
- [x] Update components to be responsive/adaptive

### PR DESCRIPTION
**Pull request by jromest**
**Description**
Implement responsive header and togglable nav in mobile screens

**Summary of Changes**
- add overflow-x hidden in the layout container
- implement a responsive header and toggle nav in mobile screens
-  set height of hero to 100vh and offset it to the header

**How to Test**
Preview it on small screens or mobile devices

**Screenshots**
screenshot mobile version header

screenshot mobile version header navigation

Closes #35

### CODE REVIEW

File: src/components/header.js [hunk 1]
```diff
@@ -11,39 +14,130 @@ const menus = [
...
-            </Link>
-          ))}
-        </Box>
-      </Flex>
-    </Container>
-  </Box>
-)
+const Header = ({ siteTitle }) => {
+  const size = useWindowSize()
+  const [isMobile, setIsMobile] = useState()
```
Comment (franreysaycon) [inline src/components/header.js]: isMobile is a derived state and an unnecessary rerender. (unless useWindowSize is just a reference)

`const isMobile = (size.width < 640) `

File: src/components/header.js [hunk 2]
```diff
@@ -11,39 +14,130 @@ const menus = [
...
+  const [showMenu, setShowMenu] = useState()
+
+  useEffect(() => {
+    if (size.width < 640) {
+      setIsMobile(true)
+      setShowMenu(false)
+    } else {
```
Comment (franreysaycon) [inline src/components/header.js]: doing the changes above and this becomes simpler. remove the else.

File: src/components/header.js [hunk 3]
```diff
@@ -11,39 +14,130 @@ const menus = [
...
+
+  useEffect(() => {
+    if (showMenu) {
+      document.body.style.overflow = "hidden"
+    } else {
+      document.body.style.overflow = "visible"
+    }
+  }, [showMenu])
+
+  const handleClickOutsideNav = e => {
```
Comment (franreysaycon) [inline src/components/header.js]: I would love it if you just convert this to a hook. but not really required :squirrel:

**Comment by jromest** [comment]
@itsdyeyson the Core Team section fix is on PR #52

**Pull request merged**